rh_aws_saml_login/_api.py

```diff
@@ -0,0 +1,32 @@
+import logging
+
+from ._consts import RH_SAML_URL, AwsRegion
+from ._core import (
+    assume_role_with_saml,
+    get_aws_account,
+    get_aws_accounts,
+    get_saml_auth,
+    is_kerberos_ticket_valid,
+)
+from ._exceptions import NoAwsAccountError, NoKerberosTicketError
+from ._models import AwsCredentials
+
+logger = logging.getLogger(__name__)
+
+
+def get_aws_credentials(
+    account_name: str,
+    saml_url: str = RH_SAML_URL,
+    session_timeout_seconds: int = 900,
+    region: str = AwsRegion.US_EAST_1,
+) -> AwsCredentials:
+    """Get AWS credentials for the given account name non-interactively."""
+    if not is_kerberos_ticket_valid():
+        raise NoKerberosTicketError
+    aws_url, saml_token = get_saml_auth(saml_url)
+    aws_accounts = get_aws_accounts(
+        aws_url, saml_token, session_timeout_seconds, region
+    )
+    if not (account := get_aws_account(aws_accounts, account_name)):
```

Comment:
Did you mean `select_aws_account`? 

```suggestion
    if not (account := select_aws_account(aws_accounts, account_name)):
```

See https://github.com/app-sre/rh-aws-saml-login/pull/150. I don't think thas `get_aws_credentials` should try to invoke CLI UI.

I acknowledge that the naming `get_aws_account` vs `select_aws_account` is suboptimal. I couldn't come up with anything better. Suggestions welcome. :slightly_smiling_face:

Reply:
> I acknowledge that the naming get_aws_account vs select_aws_account is suboptimal. I couldn't come up with anything better.

Naming things is hard, I know. I also don't have a better name. 😂